zebra/zebra_routemap.c

```diff
@@ -359,26 +359,46 @@ ALIAS (no_match_ip_address_prefix_list,
 
 DEFUN (set_src,
        set_src_cmd,
-       "set src A.B.C.D",
+       "set src (A.B.C.D|X:X::X:X)",
        SET_STR
        "src address for route\n"
        "src address\n")
 {
   struct in_addr src;
+  struct in6_addr src6;
   struct interface *pif;
+  int is_ipv4;
+  int is_ipv6;
 
-  if (inet_pton(AF_INET, argv[0], &src) <= 0)
+  is_ipv4 = inet_pton(AF_INET, argv[0], &src) > 0 ? 1 : 0;
+  is_ipv6 = inet_pton(AF_INET6, argv[0], &src6) > 0 ? 1 : 0;
+
+  if ( ! (is_ipv4 || is_ipv6))
     {
       vty_out (vty, "%% not a local address%s", VTY_NEWLINE);
       return CMD_WARNING;
     }
 
-    pif = if_lookup_exact_address (src);
-    if (!pif)
+  if (is_ipv4)
+    {
+      pif = if_lookup_exact_address (src);
+      if (!pif)
```

Comment:
if (is_ipv4) pif = ...
else pif = ...

if (!pif)
{
}

---
In reply to: [85822103](https://github.com/Azure/sonic-quagga/pull/4#discussion_r85822103) [](ancestors = 85822103)